src/commands/defi/offchain_tip_bot/airdrop.ts

```diff
@@ -0,0 +1,366 @@
+  const [authorId, amount, amountInUSD, cryptocurrency, duration, maxEntries] =
+    infos.slice(1)
+  if (authorId !== interaction.user.id) {
+    return
+  }
+  const tokenEmoji = getEmoji(cryptocurrency)
+  const endTime = dayjs()
+    .add(+duration, "second")
+    .toDate()
+  const originalAuthor = await msg.guild?.members.fetch(authorId)
+  const airdropEmbed = composeEmbedMessage(msg, {
+    title: `${defaultEmojis.AIRPLANE} An airdrop appears`,
+    description: `<@${authorId}> left an airdrop of ${tokenEmoji} **${amount} ${cryptocurrency}** (\u2248 $${roundFloatNumber(
+      +amountInUSD,
+      4
+    )})${
+      +maxEntries !== 0
+        ? ` for  ${maxEntries} ${+maxEntries > 1 ? "people" : "person"}`
+        : ""
+    }.`,
+    footer: ["Ends"],
+    timestamp: endTime,
+    originalMsgAuthor: originalAuthor?.user,
+  })
+
+  const reply = await msg
+    .edit({
+      embeds: [airdropEmbed],
+      components: [
+        new MessageActionRow().addComponents(
+          new MessageButton({
+            customId: `enter_airdrop_off-${authorId}-${duration}-${maxEntries}`,
+            label: "Enter airdrop",
+            style: "PRIMARY",
+            emoji: "🎉",
+          })
+        ),
+      ],
+    })
+    .catch(() => null)
+  if (!reply) return
+  const cacheKey = `airdrop-${reply.id}`
+  airdropCache.set(cacheKey, [], +duration)
+
+  // check airdrop expired
+  const description = `<@${authorId}>'s airdrop of ${tokenEmoji} **${amount} ${cryptocurrency}** (\u2248 $${roundFloatNumber(
+    +amountInUSD,
+    4
+  )}) `
+  await checkExpiredAirdrop(
+    reply as Message,
+    cacheKey,
+    description,
+    authorId,
+    +amount,
+    cryptocurrency,
+    duration
+  )
+}
+
+async function checkExpiredAirdrop(
+  msg: Message,
+  cacheKey: string,
+  description: string,
+  authorId: string,
+  amount: number,
+  cryptocurrency: string,
+  duration: string
+) {
+  const getParticipantsStr = (list: string[]) =>
+    list
+      .slice(0, list.length - 1)
+      .join(", ")
+      .concat(list.length === 1 ? list[0] : ` and ${list[list.length - 1]}`)
+
+  airdropCache.on("expired", async (key, participants: string[]) => {
+    if (key === cacheKey) {
+      description +=
+        participants.length === 0
+          ? "has not been collected by anyone :person_shrugging:."
+          : `has been collected by ${getParticipantsStr(participants)}!`
+
+      if (participants.length > 0 && msg.guildId) {
+        const req: OffchainTipBotTransferRequest = {
+          sender: authorId,
+          recipients: participants.map((p) => parseDiscordToken(p).value),
+          guildId: msg.guildId,
+          channelId: msg.channelId,
+          amount,
+          token: cryptocurrency,
+          each: false,
+          all: false,
+          transferType: "airdrop",
+          fullCommand: msg.content,
+          duration: +duration,
+        }
+        await Defi.OffchainDiscordTransfer(JSON.stringify(req))
```

Comment:
```suggestion
        await Defi.offchainDiscordTransfer(req)
```

`jsonFetch` có auto serialize